libs/util/AdvancedMath.arucas

```diff
@@ -0,0 +1,915 @@
+class PerlinNoise { //TODO: extend Transformation
```

Comment:
TODO?

Reply:
yeah i think that's about inheritence, the basic idea is to have the transformation methods for the different shape classes, (tho i think perlin noise is the one class that don't need it lol) Feel free to clean anything up you think needs cleaning, and if you can, do the transformation class extension too, i'm yet to work with it at all